.github/workflows/publish.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow is triggered two ways:
+#
+# 1. When a tag is created, the workflow will upload the package to
+#    test.pypi.org.
+# 2. When a release is made, the workflow will upload the package to pypi.org.
+#
+# It is done this way until PyPI has draft reviews, to allow for a two-stage
+# upload with a chance for manual intervention before the final publication.
+name: Upload package
+
+on:
+  release:
+    types: [created]
+  push:
+    tags:
+      - '*'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: "__token__"
+        TWINE_PASSWORD: ${{ secrets.TEST_PYPI_UPLOAD_TOKEN }}
+      run: |
+        if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
+          export TWINE_REPOSITORY_URL="https://test.pypi.org/legacy/"
+          export TWINE_PASSWORD="${{ secrets.TEST_PYPI_UPLOAD_TOKEN }}"
+        elif [[ "$GITHUB_EVENT_NAME" == "release" ]]; then
+          export TWINE_REPOSITORY_URL="https://upload.pypi.org/legacy/"
+          export TWINE_PASSWORD="${{ secrets.PYPI_UPLOAD_TOKEN }}"
+        else
+          echo "Unknown event name: ${GITHUB_EVENT_NAME}"
+          exit 1
+        fi
+
+        tox -e py
+        tox -e build
+        tox -e release -- --repository-url=${TWINE_REPOSITORY_URL}
```

Comment:
```suggestion
    - name: Test
      run: python -m tox -e py
    - name: Build
      run: python -m tox -e build
    - name: Publish to TestPyPI
      if: github.event_name == 'push'
      env:
        TWINE_USERNAME: __token__
        TWINE_PASSWORD: ${{ secrets.TEST_PYPI_UPLOAD_TOKEN }}
        TWINE_REPOSITORY_URL: https://test.pypi.org/legacy/
      run: python -m tox -e release -- --repository-url=${TWINE_REPOSITORY_URL}
    - name: Publish to PyPI
      if: github.event_name == 'release'
      env:
        TWINE_USERNAME: __token__
        TWINE_PASSWORD: ${{ secrets.PYPI_UPLOAD_TOKEN }}
        TWINE_REPOSITORY_URL: https://upload.pypi.org/legacy/
      run: python -m tox -e release -- --repository-url=${TWINE_REPOSITORY_URL}
```

Reply:
Alternatively:
```suggestion
    - name: Test
      run: python -m tox -e py
    - name: Build
      run: python -m tox -e build
    - name: Publish to TestPyPI
      if: github.event_name == 'push' || github.event_name == 'release'
      env:
        TWINE_USERNAME: __token__
        TWINE_PASSWORD: ${{ github.event_name == 'push' && secrets.TEST_PYPI_UPLOAD_TOKEN || secrets.PYPI_UPLOAD_TOKEN }}
        TWINE_REPOSITORY_URL: https://${{ github.event_name == 'push' && 'test' || 'upload' }}.pypi.org/legacy/'
      run: python -m tox -e release -- --repository-url=${TWINE_REPOSITORY_URL}
```

Reply:
Hm... The second one is somewhat appealing to me, but I'm not sure that using Github's pseudo-bash substitution language is any better than using a single `bash` `if/elif/else` block, and it requires duplicating the if check a bunch of times - in the "starts with" logic and in each variable you have to set.

I feel like I have other good use cases for this style so I'm glad you suggested it, but I think I'll stick with the bash script.

Reply:
https://github.com/python/tzdata/pull/18#discussion_r420787106